### DISCUSSION TITLE
Meson 0.60.2 with Visual Studio 2022?

### DISCUSSION THREAD

**Discussion by lelegard** [mesonbuild/meson#9697] [Q&A] [answered]
Hi,

On Windows 10, I installed the latest release from this repo : [meson-0.60.2-64.msi](https://github.com/mesonbuild/meson/releases/download/0.60.2/meson-0.60.2-64.msi). It was published on Nov 25, ie. quite recently.

I have upgraded Visual Studio to 2022.

On the [Builtin options](https://mesonbuild.com/Builtin-options.html) documentation page of mesonbuild.com, I read this:

> Core options ... backend {ninja, vs, vs2010, vs2012, vs2013, vs2015, vs2017, vs2019, vs2022, Xcode}

So, I type this command:

~~~
meson setup --backend vs2022 --buildtype release --default-library both <build-dir> <src-dir>
...
meson setup: error: argument --backend: invalid choice: 'vs2022' (choose from 'ninja', 'vs', 'vs2010', 'vs2012', 'vs2013', 'vs2015', 'vs2017', 'vs2019', 'Xcode')
~~~

Which meson version do we need to use vs2022? Where can we find it if the latest release of the repo does not support it?

Thanks

**Comment by eli-schwartz** [comment, marked as answer]
The meson documentation is built for git master. You will sometimes notice that parts of the documentation state that some things are new "since 0.61.0", which is not out yet.

vs2022 support was added in #9558 (the PR was submitted the day after Microsoft released it) and will be available in the next feature release, which is meson 0.61 -- unfortunately, it's not currently possible for the documentation to state which version new *values* were added.

(meson 0.60.2 is a maintenance release containing bugfixes, but not new features.)

**Comment by lelegard** [comment]
Hi @eli-schwartz 

Thanks, its perfectly makes sense. My [project](https://github.com/tsduck/tsduck) has the same characteristics, the online doc is at the repo level, sometimes ahead of the latest binary release.

When do you plan a vs2022-aware release of meson? I do not use meson for my project, but it uses [librist](https://code.videolan.org/rist/librist) which uses meson for its build. Librist delivers no binary for VS and I have to build them by myself, ending up with needing meson for VS 2022. I am sure that all Windows developers who switched to VS 2022 have the same need.

I think that the release of a new version of a major development environment such as VS is a sufficient reason for a new release of meson and all other development tools which work with VS 2019 and not with VS 2022.

Thanks

**Reply by eli-schwartz** [reply]
We release new feature versions roughly once every 3 months, give or take, so based on past release rates, I'd say we will probably make a new release mid-January.

While it is true that meson continues to work with vs2019 and in theory developers can just hold off on upgrading immediately, being held back from the latest version of developer tooling is a bit of an edge case in terms of frustration/disappointment so perhaps our release manager would feel like releasing a bit early. I don't know yet -- it can't hurt to ask him. :)

@jpakkane, any thoughts?

**Reply by jpakkane** [reply]
It should come out a few weeks after 0.60.3 is out (which should be soon, but I don't control that).
